.github/workflows/main.yml

```diff
@@ -174,6 +174,33 @@ jobs:
       - name: Test Cantera
         run: scons test
 
+  run-examples:
+    name: Run the Python examples using bash
+    runs-on: ubuntu-18.04
+    steps:
+      - uses: actions/checkout@v2
+        name: Checkout the repository
+        with:
+          submodules: recursive
+      - name: Setup Python
+        uses: actions/setup-python@v1
+        with:
+          python-version: '3.8'
+          architecture: x64
+      - name: Install Apt dependencies
+        run: sudo apt-get install libboost-dev gfortran
+      - name: Upgrade pip
+        run: python3 -m pip install -U pip setuptools wheel
+      - name: Install Python dependencies
+        run: python3 -m pip install ruamel.yaml scons numpy cython h5py pandas
+      - name: Build Cantera
+        run: python3 `which scons` build -j2
+      - name: Run the examples
+        run: |
+          find interfaces/cython/cantera/examples -type f -name '*.py' -exec python '{}' \;
```

Comment:
Replace by:
```
        run: |
          find interfaces/cython/cantera/examples -type f -name '*.py' -exec python -W error::DeprecationWarning: '{}' \;
```

Reply:
Yes, I was actually planning to set the `PYTHONWARNINGS` environment variable, but accomplishing the same thing. Thanks for the suggestion!

Reply:
Not sure whether `PYTHONWARNINGS` has the same granularity - simply setting it to `error` may catch too many exceptions.

Reply:
According to the docs, `PYTHONWARNINGS` is equivalent to the `-W` switch, and has the same syntax except that a comma-separated list of conditions is passed instead of multiple `-W` options. In any case, this is only ever intended to run on Ubuntu, but I do want to catch deprecation warnings in third-party software (e.g., NumPy), so I think catching everything until we find some exceptions is the right thing to do here.

Reply:
In the newest version of the `PYTHONWARNING` filter you implemented (`error,error::DeprecationWarning`), doesn't the first `error` already turn all warnings into exceptions? (any subsequent filter adds upon previous specifications, so there's nothing new added?) Regarding the environment variable, I wasn't sure as my attempts to use it with scons were not successful, but the [docs](https://docs.python.org/3/library/warnings.html) are indeed quite clear ...

Reply:
> In the newest version of the `PYTHONWARNINGS` filter you implemented (`error,error::DeprecationWarning`), doesn't the first `error` already turn all warnings into exceptions?

I think it should, that was my attempt to debug something where I thought there was a deprecation warning that wasn't being converted. It could probably be removed.